src/sage/schemes/elliptic_curves/gal_reps_number_field.py

```diff
@@ -402,6 +402,369 @@ def reducible_primes(self):
+            sage: G.is_nonsplit_normaliser(5)
+            True
+            sage: E = EllipticCurve([0, -1, 1, -5373758, -9740388623]) # '15341b1'
+            sage: G = GaloisRepresentation(E)
+            sage: G.is_nonsplit_normaliser(7)
+            True
+
+        TESTS::
+
+            sage: from sage.schemes.elliptic_curves.gal_reps_number_field import GaloisRepresentation
+            sage: E = EllipticCurve('11a1')
+            sage: G = GaloisRepresentation(E)
+            sage: G.is_nonsplit_normaliser(4)
+            Traceback (most recent call last):
+            ...
+            NotImplementedError: p = 4 should be prime, either 3, 5, or 7
+            sage: G.is_nonsplit_normaliser(x)
+            Traceback (most recent call last):
+            ...
+            ValueError: p = x should be a prime integer, either 3, 5, or 7
+            sage: G.is_nonsplit_normaliser(11)
+            Traceback (most recent call last):
+            ...
+            NotImplementedError: Only implemented for p<=7
+        """
+        if p == 2:
+            raise ValueError("Normaliser of nonsplit Cartan not defined for p=2")
+        j = self.E.j_invariant()
+        x = polygen(j.parent())
+        pol = None
+        if p == 3:
+            pol = x**3 - j
+        if p == 5:
+            pol = 5**4 * (2*x+1) * (x+1)**3 * (6*x**2+21*x+19)**3 - j * (x**2+x-1)**5
+        if p == 7:
+            pol = ((4*x**2+5*x+2) * (x**2+3*x+4) * (x**2+10*x+4) * (3*x+1))**3 - j * (x**3+x**2-2*x-1)**7
+        if pol:
+            return bool(pol.roots())
+
+        try:
+            if ZZ(p).is_prime():
+                raise NotImplementedError("Only implemented for p<=7")
+            else:
+                raise NotImplementedError("p = {} should be prime, either 3, 5, or 7".format(p))
+        except TypeError:
+            raise ValueError("p = {} should be a prime integer, either 3, 5, or 7".format(p))
+
+    def is_nonsplit(self, p):
+        r"""
+        Return ``True`` if the projective mod-`p` image is contained in a nonsplit Cartan subgroup.
+
+        INPUT:
+
+        - ``p`` (integer) -- a prime number
+
+        OUTPUT:
+
+        (boolean) ``True`` if the projective mod-`p` image is
+        contained in a nonsplit Cartan subgroup, else ``False``.
+
+        ALGORITHM:
+
+        This method is currently only implemented for primes `p` for
+        which the modular curve `X_{\text{ns}}(p) has genus 0 and has
+        a rational point, namely `p=2` only.  (For `p=3,5` the genus
+        is zero but there are no rational points.)  In this cases we
+        use an explicit formula for the map from the modular curve to
+        the `j`-line, checking that the curve's `j`-invariant is in
+        the image.  For `p=2` this is simply that `j-1728` must be a
+        square.
+        """
+        j = self.E.j_invariant()
+        if p == 2:
+            return (j-1728).is_square()
+
+        raise NotImplementedError("Only implemented for p==2")
+
```

Comment:
Error messages should rather start with a smallcap letter, i.e. not with a capital. This is python convention, don't ask me why..

Reply:
OK, I am editing now, how will I know you have finished?

Reply:
> OK, I am editing now, how will I know you have finished?

These are submitted in a batch AFAIK. So I think @fchapoton finished looking at things.

Reply:
yes, sorry, busy with other matters. I have no other comment

Reply:
> yes, sorry, busy with other matters. I have no other comment

thanks -- as always -- for your attention to detail